File: postgres_39/all_struct_with_default/SeisProcSet.go
package dto

type Seis_proc_set struct {
	Seis_set_subtype   string   `json:"seis_set_subtype" default:"source"`
	Seis_proc_set_id   string   `json:"seis_proc_set_id" default:"source"`
	Active_ind         *string  `json:"active_ind" default:""`
	Area_id            *string  `json:"area_id" default:""`
	Area_type          *string  `json:"area_type" default:""`
	Description        *string  `json:"description" default:""`
	Effective_date     *string  `json:"effective_date" default:""`
	End_date           *string  `json:"end_date" default:""`
	Expiry_date        *string  `json:"expiry_date" default:""`
	Max_latitude       *float64 `json:"max_latitude" default:""`
	Max_longitude      *float64 `json:"max_longitude" default:""`
	Min_latitude       *float64 `json:"min_latitude" default:""`
	Min_longitude      *float64 `json:"min_longitude" default:""`
	Objective          *string  `json:"objective" default:""`
	Original_proc_ind  *string  `json:"original_proc_ind" default:""`
	Ppdm_guid          *string  `json:"ppdm_guid" default:""`
	Processed_for      *string  `json:"processed_for" default:""`
	Processing_company *string  `json:"processing_company" default:""`
	Processing_name    *string  `json:"processing_name" default:""`
	Process_status     *string  `json:"process_status" default:""`
	Proc_set_type      *string  `json:"proc_set_type" default:""`
	Remark             *string  `json:"remark" default:""`
	Reprocessed_ind    *string  `json:"reprocessed_ind" default:""`
	Source             *string  `json:"source" default:""`
	Start_date         *string  `json:"start_date" default:""`
	Row_changed_by     *string  `json:"row_changed_by" default:""`
	Row_changed_date   *string  `json:"row_changed_date" default:""`
	Row_created_by     *string  `json:"row_created_by" default:""`
	Row_created_date   *string  `json:"row_created_date" default:""`
	Row_effective_date *string  `json:"row_effective_date" default:""`
	Row_expiry_date    *string  `json:"row_expiry_date" default:""`
	Row_quality        *string  `json:"row_quality" default:""`
}

File: create_tables/seis_proc_set.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table seis_proc_set';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """CREATE TABLE seis_proc_set(
        SEIS_SET_SUBTYPE VARCHAR(30) NOT NULL,
        SEIS_PROC_SET_ID VARCHAR(40) NOT NULL,
        ACTIVE_IND VARCHAR(1),
        AREA_ID VARCHAR(40),
        AREA_TYPE VARCHAR(40),
        DESCRIPTION VARCHAR(2000),
        EFFECTIVE_DATE TIMESTAMP(0),
        END_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        MAX_LATITUDE DECIMAL(14,9),
        MAX_LONGITUDE DECIMAL(14,9),
        MIN_LATITUDE DECIMAL(14,9),
        MIN_LONGITUDE DECIMAL(14,9),
        OBJECTIVE VARCHAR(2000),
        ORIGINAL_PROC_IND VARCHAR(1),
        PPDM_GUID VARCHAR(38),
        PROCESSED_FOR VARCHAR(40),
        PROCESSING_COMPANY VARCHAR(40),
        PROCESSING_NAME VARCHAR(255),
        PROCESS_STATUS VARCHAR(40),
        PROC_SET_TYPE VARCHAR(40),
        REMARK VARCHAR(2000),
        REPROCESSED_IND VARCHAR(1),
        SOURCE VARCHAR(40),
        START_DATE TIMESTAMP(0),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")